backend/src/utils/adminUtils.ts

```diff
@@ -0,0 +1,88 @@
+import { CustomObjectsApi } from '@kubernetes/client-node';
+import { KubeFastifyInstance } from '../types';
+import { getGroupsConfigMapName, getAdminGroups, getGroup } from './groupsUtils';
+
+const SYSTEM_AUTHENTICATED = 'system:authenticated';
+
+export const getGroupsConfig = async (
+  fastify: KubeFastifyInstance,
+  customObjectApi: CustomObjectsApi,
+  userName: string,
+): Promise<boolean> => {
+  try {
+    const coreV1Api = fastify.kube.coreV1Api;
+    const groupConfig = await getGroupsConfigMapName(coreV1Api, fastify.kube.namespace);
+    const adminGroups = await getAdminGroups(coreV1Api, fastify.kube.namespace, groupConfig);
+
+    const adminGroupsList = adminGroups.split(',');
+
+    if (adminGroupsList.includes(SYSTEM_AUTHENTICATED)) {
+      throw new Error('It is not allowed to set system:authenticated or as admin group.');
+    } else {
+      return await checkUserInGroups(fastify, customObjectApi, adminGroupsList, userName);
+    }
+  } catch (e) {
+    fastify.log.error(e.toString());
+    return false;
+  }
+};
+
+export const isUserAdmin = async (
```

Comment:
I made my function non-async as I wanted to loop this check over multiple users -- I think it'll be easier to align to what I did then rewriting existing code to match

Reply:
It needs to be asyn, since one of the edge-cases now is that cluster-admins are considered admins too, so we need to check the rolebindings.